### PR TITLE
#12010 Return time editor

### PR DESCRIPTION
**Pull request by Destrolaric**
Added new editor for time and Date in Data Editor, it allows user to edit time without needing to write in string manually. It allows setting dates by using the SWT calendar. Also can be used in panel mode. 

image
image
image
We can set default edit mode in preferences
image

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/CustomTimeEditor.java [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
  * DBeaver - Universal Database Manager
- * Copyright (C) 2010-2021 DBeaver Corp and others
+ * Copyright (C) 2010-2017 Serge Rider (serge@jkiss.org)
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/CustomTimeEditor.java]: Copyright (C) 2010-2021 DBeaver Corp and others

Comment (Destrolaric) [reply plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/CustomTimeEditor.java]: Fixed

**Comment by Destrolaric** [comment]
https://user-images.githubusercontent.com/20579256/147641522-947699f0-4adc-4288-9156-f02391052fa4.mp4

File: plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/CustomTimeEditor.java [hunk 2]
```diff
@@ -16,62 +16,313 @@
...
+
+        }
+        if (!timeEditor.isDisposed()) {
+            timeEditor.addTraverseListener(e -> timeEditor.setFocus());
+            timeEditor.setTime(calendar.get(Calendar.HOUR_OF_DAY), calendar.get(Calendar.MINUTE), calendar.get(Calendar.SECOND));
+
+            try {
+                millis = calendar.get(Calendar.MILLISECOND);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                //Calendar doesn't have any way to
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/CustomTimeEditor.java:279]: I have a feeling that this sentence is not finished

**Pull request merged**